Exiled.API/Features/Player.cs

```diff
@@ -953,19 +954,18 @@ public bool BadgeHidden
         public bool CanSendInputs => Role.FirstPersonController.FpcModule.LockMovement;
 
         /// <summary>
-        /// Gets a <see cref="Player"/> <see cref="IEnumerable{T}"/> of spectators that are currently spectating this <see cref="Player"/>.
+        /// Gets a <see cref="Player"/> <see cref="IReadOnlyCollection{T}"/> of spectators that are currently spectating this <see cref="Player"/>.
         /// </summary>
-        public IEnumerable<Player> CurrentSpectatingPlayers
+        public IReadOnlyList<Player> CurrentSpectatingPlayers
         {
             get
             {
-                foreach (Player player in List)
+                List<Player> spectators = new();
+                SpectatorNetworking.ForeachSpectatorOf(ReferenceHub, (spectator) =>
                 {
-                    if (player == this || player.Role is not SpectatorRole role || role.SpectatedPlayer != this)
```

Comment:
Can't we just use `player.ReferenceHub.IsSpectatedBy(...)` instead of creating a new List every time someone calls this property?

Reply:
Wouldn't that still require a list?

Reply:
Why not just using instead of Foreach ?

Reply:
Since it's an IEnumerable yeah, we could just do something like `Player.List.Where(player => ReferenceHub.IsSpectatedBy(player.ReferenceHub))`